### PR TITLE
Initial import

### PR DESCRIPTION
**Pull request by justbitdev**
### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [x] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/cms.tf [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+resource "azurerm_role_assignment" "app_service_container_registry" {
+  scope                            = azurerm_container_registry.container_registry.id
+  role_definition_name             = "AcrPull"
+  principal_id                     = module.portal_backend.principal_id
+  skip_service_principal_aad_check = true
+}
+
+data "azurerm_key_vault_secret" "cms_db_password" {
+  depends_on = [azurerm_key_vault_access_policy.terraform_cloud_policy]
+  #count        = var.agid_spid_public_cert != null ? 1 : 0
```
Comment (uolter) [inline src/cms.tf]:
```suggestion
  
```

File: src/cms.tf [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  #linux_fx_version = format("DOCKER|%s/ccorp-site:%s", azurerm_container_registry.container_registry.login_server, "latest")
+  linux_fx_version = "DOCKER|nginx"
+  #   linux_fx_version = format("DOCKER|ppareg.azurecr.io/corporate-site-backend:%s", "latest")
+  # linux_fx_version = "NODE|14-lts"
+
+  always_on = "true"
+
+  allowed_subnets = [module.subnet_wp.id, module.subnet_db.id, module.subnet_public.id]
+  allowed_ips     = ["0.0.0.0/0"]
```
Comment (uolter) [inline src/cms.tf]:
```suggestion
  allowed_ips     = []
```

File: src/database.tf [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  tags = var.tags
+
+}
+
+resource "azurerm_mysql_firewall_rule" "mysql_firewall_rule_public" {
+  name                = "azure"
+  resource_group_name = azurerm_resource_group.rg_db.name
+  server_name         = azurerm_mysql_server.mysql_server.name
+  start_ip_address    = "0.0.0.0"
```
Comment (uolter) [inline src/database.tf:45]: Does this mean the database is open to the public internet? 
I'm expecting that only the cms subnet can access the database.

File: src/database.tf [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+resource "azurerm_mysql_database" "mysql_database" {
+  name                = var.database_name
+  resource_group_name = azurerm_resource_group.rg_db.name
+  server_name         = azurerm_mysql_server.mysql_server.name
+  charset             = var.db_charset
+  collation           = var.db_collation
+}
+
+# resource "azurerm_private_endpoint" "mysql_private_endpoint" {
```
Comment (uolter) [inline src/database.tf:95]: Why are you removing the private endpoint? I think this is a best practice to access the database server via private endpoint. It is what we are doing in this project too: 
https://github.com/pagopa/cstar-infrastructure/tree/main/src

File: src/env/dev/terraform.tfvars [hunk 5]
```diff
@@ -0,0 +1,22 @@
+env_short = "d"
+env_long  = "development"
+prefix    = "ppacorpsite"
+tags = {
+  CreatedBy   = "Terraform"
+  Environment = "Dev"
+  Owner       = "site-corporate"
+  Source      = "https://github.com/pagopa/corporate-site-infrastructure"
+  CostCenter  = ""
```
Comment (uolter) [inline src/env/dev/terraform.tfvars:10]: @vito80ba do we you know the cost center here?
Puo essere TS300 - PRODOTTI E SERVIZI

File: src/modules/storage_account/main.tf [hunk 6]
```diff
@@ -0,0 +1,113 @@
+
+
+resource "azurerm_storage_account" "storage_account" {
```
Comment (uolter) [inline src/modules/storage_account/main.tf]:
```suggestion
resource "azurerm_storage_account" "this" {
```

File: src/modules/cdn_endpoint/main.tf [hunk 7]
```diff
@@ -0,0 +1,94 @@
+
+resource "azurerm_cdn_endpoint" "cdn_endpoint" {
```
Comment (uolter) [inline src/modules/cdn_endpoint/main.tf:2]:
```suggestion
resource "azurerm_cdn_endpoint" "this" {
```

File: src/network.tf [hunk 8]
```diff
@@ -0,0 +1,82 @@
+resource "azurerm_resource_group" "rg_vnet" {
+  name     = format("%s-vnet-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+## Network security groups:
+### database
+resource "azurerm_network_security_group" "db_nsg" {
```
Comment (uolter) [inline src/network.tf]: I think you can remove this nsg

File: src/network.tf [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  location            = azurerm_resource_group.rg_vnet.location
+  resource_group_name = azurerm_resource_group.rg_vnet.name
+  address_space       = var.cidr_vnet
+
+  tags = var.tags
+
+}
+
+module "subnet_db" {
+  source                                         = "./modules/subnet"
```
Comment (uolter) [inline src/network.tf:23]: why you didn't use existing modules in https://github.com/pagopa/azurerm ? 
We already have modules for vnet subnet etc

File: src/storage.tf [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  name     = format("%s-storage-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+module "storage_account" {
+  source = "./modules/storage_account"
+
+  name            = replace(format("%s-sa", local.project), "-", "")
```
Comment (uolter) [inline src/storage.tf:11]: the sa suffix stands for? I mean what's the purpose of this storage?

File: src/variables.tf [hunk 11]
```diff
@@ -12,9 +12,232 @@ variable "env_short" {
...
+variable "db_create_mode" {
+  type        = string
+  description = "The creation mode. Can be used to restore or replicate existing servers."
+  default     = "Default"
+}
+
+variable "db_public_network_access_enabled" {
+  type        = bool
+  description = "Whether or not public network access is allowed for this server."
+  default     = false
```
Comment (uolter) [inline src/variables.tf:84]: I would have put true .... as per default value .

**Review by uolter — CHANGES_REQUESTED** [review]

File: src/cms.tf [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  #linux_fx_version = format("DOCKER|%s/ccorp-site:%s", azurerm_container_registry.container_registry.login_server, "latest")
+  linux_fx_version = "DOCKER|nginx"
+  #   linux_fx_version = format("DOCKER|ppareg.azurecr.io/corporate-site-backend:%s", "latest")
+  # linux_fx_version = "NODE|14-lts"
+
+  always_on = "true"
+
+  allowed_subnets = [module.subnet_wp.id, module.subnet_db.id, module.subnet_public.id]
```
Comment (uolter) [inline src/cms.tf]:
```suggestion
  allowed_subnets = [module.subnet_wp.id, module.subnet_db.id]
```

**Review by uolter — CHANGES_REQUESTED** [review]

File: src/storage.tf (hunk 10 above)
Comment (nottix) [reply src/storage.tf:11]: sa: storage_account.
We need this storage to keep all media objects of CMS (images, pdfs, etc.)

File: src/cms.tf (hunk 2 above)
Comment (nottix) [reply src/cms.tf]: we need this rule for now to access CMS backoffice. We will put a basic authentication protection on top to restrict access. Is it ok?

File: src/storage.tf (hunk 10 above)
Comment (uolter) [reply src/storage.tf:11]: can we rename it something like assets?

File: src/cms.tf (hunk 2 above)
Comment (uolter) [reply src/cms.tf]: why don't you put an application gateway (ALB in AWS garden)  in front of the cms instead of exposing the app service itself?

**Review by uolter — DISMISSED** [review]

**Review by nottix — DISMISSED** [review]

**Review by pasqualedevita — APPROVED** [review]

**Comment by pasqualedevita** [comment]
We'll merge into main branch but there are some fixes to do (ie use azurerm modules)

**Pull request merged**